File: C++/题目1470：调整方阵.cpp

#include <stdio.h>
const int maxn = 12;
int matrix[maxn][maxn];
int n,i,j;

int main(){
    while(scanf("%d",&n) != EOF){
		for(i = 0; i < n; i++){
			for(j = 0; j < n; j++){
				scanf("%d",&matrix[i][j]);
			}
		}
		for (j = 0; j < n; j++) {
			int maxValue = matrix[j][j];
            int maxCol = j;
            for (i = j+1; i < n; i++) {
				if (matrix[i][j] > maxValue) {
					maxValue = matrix[i][j];
                    maxCol = i;
                }
            }
            for (i = 0; i < n; i++) {
				int temp = matrix[j][i];
                matrix[j][i] = matrix[maxCol][i];
                matrix[maxCol][i] = temp;
            }
        }
        for (int i = 0; i < n; i++) {
			for (int j = 0; j < n-1; j++) {
				printf("%d ",matrix[i][j]);
            }
			printf("%d",matrix[i][n-1]);
            printf("\n");
		}
    }
    return 0;
}

/**************************************************************
	Problem: 1470
	User: wangzhenqing
	Language: C++
	Result: Accepted
	Time:0 ms
	Memory:1020 kb
****************************************************************/


                        

File: Java/题目1470：调整方阵.java

// 题目1470：调整方阵

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:55
 * @url:http://ac.jobdu.com/problem.php?pid=1470
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.StreamTokenizer;

public class Main {
    /*
     * 1053
     */
    public static void main(String[] args) throws IOException {
        StreamTokenizer st = new StreamTokenizer(new BufferedReader(
                new InputStreamReader(System.in)));
        while (st.nextToken() != StreamTokenizer.TT_EOF) {
            int n = (int) st.nval;
            int matrix[][] = new int[n][n];
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    st.nextToken();
                    matrix[i][j] = (int) st.nval;
                }
            }
            for (int j = 0; j < n; j++) {
                int maxValue = matrix[j][j];
                int maxCol = j;
                for (int i = j + 1; i < n; i++) {
                    if (matrix[i][j] > maxValue) {
                        maxValue = matrix[i][j];
                        maxCol = i;
                    }
                }
                for (int i = 0; i < n; i++) {
                    int temp = matrix[j][i];
                    matrix[j][i] = matrix[maxCol][i];
                    matrix[maxCol][i] = temp;
                }
            }
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n - 1; j++) {
                    System.out.print(matrix[i][j] + " ");
                }
                System.out.print(matrix[i][n - 1]);
                System.out.println();
            }
        }
    }

}
/**************************************************************
 Problem: 1470
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:100 ms
 Memory:14868 kb
 ****************************************************************/




